tests/db/test_v6_to_v7.py

```diff
@@ -0,0 +1,145 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import datetime
+import os
+import uuid
+from dataclasses import dataclass
+from typing import AsyncIterator, Awaitable, Callable, List
+
+import pytest
+from asyncpg import Connection, Record
+from asyncpg.cursor import Cursor
+
+from db.common import PGRestore
+from inmanta import data
+from inmanta.server.services.databaseservice import DatabaseService
+
+
+@pytest.fixture
+async def migrate_v6_to_v7(
+    hard_clean_db, hard_clean_db_post, postgresql_client: Connection, async_finalizer, server_config
+) -> AsyncIterator[Callable[[], Awaitable[None]]]:
+    """
+    Performs a v6 database restore and yields a function that migrates to v7 when called.
+
+    Word of caution:
+    This fixture differs from similar ones in previous version migration tests in the following ways:
+    - It yields a migration function instead of migrating before yielding. This to allow tests to set up desired preconditions.
+    - It only starts the database slice, not the full server. This to make sure agent instance documents after yielding are only
+        affected by the migration, not by any agent management of the server. This prevents agent instance documents expiring
+        when starting the server.
+    """
+    # Get old tables
+    with open(os.path.join(os.path.dirname(__file__), "dumps/v6.sql"), "r") as fh:
+        await PGRestore(fh.readlines(), postgresql_client).run()
+
+    db_service: DatabaseService = DatabaseService()
+
+    # When the bootloader is started, it also executes the migration to v7
+    yield db_service.start
+    await db_service.stop()
+
+
+@dataclass
+class AgentInstanceCount:
+    """
+    Helper class for describing agent instance table preconditions for a specific tid, process, name combination.
+    """
+
+    tid: uuid.UUID
+    process: uuid.UUID
+    name: str
+    count: int
+    active_count: int
+
+
+@pytest.mark.asyncio(timeout=20)
+async def test_unique_agent_instances(migrate_v6_to_v7: Callable[[], Awaitable[None]], postgresql_client: Connection) -> None:
```

Comment:
Also test if the agentinstance included in the db dump are migrated correctly.

Reply:
If I understand you correctly this is done [here](https://github.com/inmanta/inmanta/pull/2427/files/6ccdbfbe6a7ee089c83ec7b89b33b144edd87846#diff-31c0bcc09f912ffbe2d8a9854a5b3999cd4011896882a55afd89d2aadb822538R127).